community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/ZonedTimeSchemaKey.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.index.schema;
+
+import java.time.ZoneOffset;
+
+import org.neo4j.values.storable.TimeValue;
+import org.neo4j.values.storable.Value;
+
+import static java.lang.String.format;
+
+/**
+ * Includes value and entity id (to be able to handle non-unique values). A value can be any {@link TimeValue}.
+ *
+ * With these keys the TimeValues are sorted by UTC time of day, and then by time zone.
+ */
+class ZonedTimeSchemaKey extends TemporalSchemaKey implements ComparableNativeSchemaKey<ZonedTimeSchemaKey>
+{
+    static final int SIZE =
+            Long.BYTES +    /* nanosOfDayUTC */
+            Integer.BYTES + /* zoneOffsetSeconds */
+            Long.BYTES;     /* entityId */
+
+    long nanosOfDayUTC;
+    int zoneOffsetSeconds;
+
+    @Override
+    public Value asValue()
+    {
+        return TimeValue.time( nanosOfDayUTC, ZoneOffset.ofTotalSeconds( zoneOffsetSeconds ) );
+    }
+
+    @Override
+    public void initAsLowest()
+    {
+        super.initAsLowest();
+        nanosOfDayUTC = Long.MIN_VALUE;
+        zoneOffsetSeconds = Integer.MIN_VALUE;
+    }
+
+    @Override
+    public void initAsHighest()
+    {
+        super.initAsHighest();
+        nanosOfDayUTC = Long.MAX_VALUE;
+        zoneOffsetSeconds = Integer.MAX_VALUE;
+    }
+
+    @Override
+    public int compareValueTo( ZonedTimeSchemaKey other )
```

Comment:
I THINK you are doing the same as the `compareTo` method in Java's `OffsetTime.java` which is what  we use from the `TimeValue` code. But you have implemented it quite differently. Can you double check that your method does the same thing as the Java one?